packages/core/src/model.ts

```diff
@@ -163,6 +163,24 @@ export class ChartModel {
 			displayData,
 			(datum) => datum[domainIdentifier]
 		).keys();
+
+		const axisPosition = this.services.cartesianScales.domainAxisPosition; 
+		const scaleType = options.axes[axisPosition].scaleType;
+
+		// Sort keys
+		if (scaleType === ScaleTypes.TIME) {
+			stackKeys.sort((a, b) => {
+				const dateA: any = new Date(a);
+				const dateB: any = new Date(b);
+				return dateA - dateB;
+			})
```

Comment:
optional

```suggestion
			stackKeys.sort((a, b) => new Date(a) - new Date(b));
```

otherwise

```suggestion
			stackKeys.sort((a, b) => {
				const dateA: any = new Date(a);
				const dateB: any = new Date(b);
				return dateA - dateB;
			});
```

Reply:
I tried that, but Type error.

Reply:
Okay, then just the semicolon then :)

Reply:
`(a: any, b: any) => new Date(a) - new Date(b)`

would be the best way

Reply:
The left and right side of arithmetic operator cannot be Date, so it requires `new Date(a): any` and `new Date(b): any`.

Reply:
ok

can you re-push the change please? I commitd doni's suggestion already

Reply:
No worries, I did.